### PR TITLE
Switch back to w2dynamics upstream repository

### PR DESCRIPTION
**Pull request by hmenke**
We have to pin Python to 3.11 in CI, because w2dynamics depends on distutils which was removed in Python 3.12.

Removing distutils from w2dynamics would be a larger undertaking since it would require reworking the entire f2py integration.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -77,11 +83,6 @@ jobs:
       run:
         echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV
 
-    - name: add clang cxxflags
-      if: ${{ contains(matrix.cxx, 'clang') }}
-      run:
-        echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV
-
```
Comment (hmenke) [inline .github/workflows/build.yml:84]: This block was duplicated.

**Review by the-hampel — APPROVED** [review]
Thank you @hmenke for the fixes. I agree we should limit the w2dyn interface for now to Python 3.11 maximum until https://github.com/w2dynamics/w2dynamics/pull/25 is merged into w2dynamics. Then we should remove the constrain again.

**Comment by the-hampel** [comment]
@hmenke @HugoStrand @Wentzell , 
for python version <3.12 this works now just fine, i.e. using the master branch of w2dynamics. However, for py ver 3.12 support they would need to merge https://github.com/w2dynamics/w2dynamics/pull/25 . I would vote for merging this now. If w2dynamics merges the PR there are no further changes necessary here to enable py 3.12 support.

**Comment by Wentzell** [comment]
I have updated w2dynamics to the very latest commit https://github.com/w2dynamics/w2dynamics/commit/7364c002c1c1e2f5aaa27388018c443ca11338f4 on unstable

I have also changed the cmake logic to now use `FetchContent` instead of `ExternalProject`

**Comment by Wentzell** [comment]
CI is passing also without fixing the Python Version in ghactions.
C.f. https://github.com/TRIQS/w2dynamics_interface/actions/runs/8427811994

**Pull request closed without merging**